contracts/DCAStrategies/DCAStrategies/DCAStrategiesPermissionsHandler.sol

```diff
@@ -95,18 +107,17 @@ abstract contract DCAStrategiesPermissionsHandler is IDCAStrategiesPermissionsHa
   }
 
   function _setPermissions(uint256 _id, IDCAStrategies.PermissionSet[] calldata _permissions) internal {
-    // uint248 _blockNumber = uint248(_getBlockNumber());
-    // for (uint256 i; i < _permissions.length; i++) {
-    //   if (_permissions[i].permissions.length == 0) {
-    //     delete tokenPermissions[keccak256(abi.encodePacked(_id, _permissions[i].operator))];
-    //   } else {
-    //     // TODO: remove this comment when adding permission math library
-    //     tokenPermissions[keccak256(abi.encodePacked(_id, _permissions[i].operator))] = TokenPermission({
-    //       permissions: _permissions[i].permissions.toUInt8(),
-    //       lastUpdated: _blockNumber
-    //     });
-    //   }
-    // }
+    uint248 _blockNumber = uint248(_getBlockNumber());
+    for (uint256 i; i < _permissions.length; i++) {
```

Comment:
Look at what Sam did in `_setPermissions` [here](https://github.com/Mean-Finance/dca-v2-core/pull/232/files). By copying the struct to memory, you can avoid accessing the array multiple times, making it cheaper